### PR TITLE
Add python unittests

### PR DESCRIPTION
**Pull request by rajivsam**
refactored old test cases with python's unit test

### CODE REVIEW

File: arangopipe/Dockerfile_Single_Image [hunk 1]
```diff
@@ -7,7 +7,7 @@ RUN apt-get update
 RUN apt-get install -y python-pip
 #RUN pip install --upgrade pip
 RUN pip install conda mlflow hyperopt sklearn2 jsonpickle python-arango==4.4.0
-RUN pip install -i https://test.pypi.org/simple/ arangopipe==0.0.35
+RUN pip install -i https://test.pypi.org/simple/ arangopipe
```
Comment (joerg84) [inline arangopipe/Dockerfile_Single_Image]: Why did you change this?

File: arangopipe/tests/.spyproject/workspace.ini [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[workspace]
+restore_data_on_startup = True
```
Comment (joerg84) [inline arangopipe/tests/.spyproject/workspace.ini]: what is .spyproject? do we need this checked in?

File: arangopipe/tests/.spyproject/workspace.ini [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+save_data_on_exit = True
+save_history = True
+save_non_project_files = False
+
+[main]
+version = 0.1.0
+recent_files = ['/home/admin2/arangopipe/git_wd2/arangopipe/tests/arangopipe_tests.py', '/home/admin2/arangopipe/git_wd2/arangopipe/arangopipe/arangopipe_admin_api.py', '/home/admin2/arangopipe/git_wd2/arangopipe/tests/arangopipe_testcases.py', '/home/admin2/arangopipe/git_wd2/arangopipe/arangopipe/arangopipe_api.py']
```
Comment (joerg84) [inline arangopipe/tests/.spyproject/workspace.ini]: this looks like it shouldn't be checked in...

File: arangopipe/tests/.spyproject/workspace.ini (hunk 2 above)
Comment (rajivsam) [reply arangopipe/tests/.spyproject/workspace.ini]: removed it, we dont need it

File: arangopipe/tests/.spyproject/workspace.ini (hunk 3 above)
Comment (rajivsam) [reply arangopipe/tests/.spyproject/workspace.ini]: yes, removed it. We don't need it

File: arangopipe/tests/arangopipe_testcases.py [hunk 4]
```diff
@@ -1,170 +1,324 @@
...
+    def dataset_shift_negative(self):
+        ds_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "covariate_shift/cal_housing.csv")
+        df = pd.read_csv(ds_path)
+        req_cols = df.columns.tolist()
+        df = df[req_cols]
+        df1 = df.query("lat <= -119")
+        df2 = df1.copy()
+        rfd = RF_DatasetShiftDetector()
+        score = rfd.detect_dataset_shift(df1, df2)
+        print ("Detaset shift score : ", score)
```
Comment (joerg84) [inline arangopipe/tests/arangopipe_testcases.py]: This test still doesn't test... when would it fail? I would expect some if (delta> threshold) : fail_test

File: arangopipe/tests/arangopipe_testcases.py [hunk 5]
```diff
@@ -1,170 +1,324 @@
...
+    def dataset_shift_positive(self):
+        ds_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "covariate_shift/cal_housing.csv")
+        df = pd.read_csv(ds_path)
+        req_cols = df.columns.tolist()
+        df = df[req_cols]
+        df1 = df.query("lat <= -119")
+        df2 = df.query("lat > -119")
+        rfd = RF_DatasetShiftDetector()
+        score = rfd.detect_dataset_shift(df1, df2)
+        print ("Detaset shift score : ", score)
```
Comment (joerg84) [inline arangopipe/tests/arangopipe_testcases.py]: same

File: arangopipe/setup.py [hunk 6]
```diff
@@ -11,7 +11,7 @@
 # This call to setup() does all the work
 setup(
     name="arangopipe",
-    version="0.0.35",
+    version="0.0.37",
```
Comment (joerg84) [inline arangopipe/setup.py]: What is the reason to skip 36?

File: arangopipe/tests/arangopipe_testcases.py (hunk 5 above)
Comment (joerg84) [reply arangopipe/tests/arangopipe_testcases.py]: As you only use this once I would put it directly into the test below...
